test/mocks.ts

```diff
@@ -127,3 +127,49 @@ export function expectInvalidProperties(
     expect(Object.values(validationErrors as any)?.[0]).to.include(expectedError);
   }
 }
+
+export const mockUserRole = (allowed_roles:  string[]): ContactProperty => ({
+  friendly_name: 'Role',
+  property_name: 'role',
+  parameter: allowed_roles,
+  required: true
+});
+
+export const mockValidMultipleRolesContactType = (propertyType, propertyValidator: string | string[] | undefined): ContactType => {
+  const baseMock = mockValidContactType(propertyType, propertyValidator);
+
+  return {
+    ...baseMock,
+    user_role: mockUserRole(['role1', 'role2']),
+  };
+};
+
+export const mockSimpleMultipleRolesContactType = (
+  propertyType,
+  propertyValidator: string | string[] | undefined,
+  errorDescription?: string,
+  allowedRoles?: string[],
+) : ContactType => {
+  const mockedProperty = mockProperty(propertyType, propertyValidator);
+  mockedProperty.errorDescription = errorDescription;
+  return {
+    name: 'contacttype-name',
+    friendly: 'friendly',
+    contact_type: 'contact-type',
+    user_role: mockUserRole(allowedRoles ?? ['role1', 'role2']),
```

Comment:
better to use default parameters on line 51 instead of providing a default here?  `allowedRoles: string[] = ['role1', 'role2']` ?